### PR TITLE
refactored remote repository to use inputted url

### PR DESCRIPTION
**Pull request by whaley-ethan**
## Summary
The `remote-repository-testing` branch refactors the remote repository to allow setting of a custom url, and adds tests for this functionality

## Future Work:
-test the edge case of a non-existent/ nonAPI URL

### CODE REVIEW

File: GuildedRoseLLC/Items/RemoteItemRepository.swift [hunk 1]
```diff
@@ -2,12 +2,21 @@ import UIKit
 
 public class RemoteItemRepository: ItemRepository {
     
-    public init() {}
+    public static var stagingAPI: String = "https://floating-spire-59497.herokuapp.com/api/v1/items"
+    public var url: String
+    
+    public init() {
+        url = RemoteItemRepository.stagingAPI
```
Comment (cmvandrevala) [inline GuildedRoseLLC/Items/RemoteItemRepository.swift:9]: Comment, no changes needed: If I understand this correctly, you are setting the url to the staging API url by default, but you have the option to change it via `setURL`, correct? I think this is ok for now. Just note that usually, details like URLs, keys, usernames, passwords, etc. are often stored in config files - sometimes these files are even encrypted. This setup might make it difficult to deploy one instance of the app to staging and one instance of the app to a production environment

**Review by cmvandrevala — APPROVED** [review]

File: GuildedRoseLLC/Items/RemoteItemRepository.swift (hunk 1 above)
Comment (elijahsullivan) [reply GuildedRoseLLC/Items/RemoteItemRepository.swift:9]: +1 to this and if your configured URLs are not malformed I would expect you to be able to remove the `guard let requestUrl = ...`.

File: GuildedRoseLLC/Items/RemoteItemRepository.swift [hunk 2]
```diff
@@ -2,12 +2,21 @@ import UIKit
 
 public class RemoteItemRepository: ItemRepository {
     
-    public init() {}
+    public static var stagingAPI: String = "https://floating-spire-59497.herokuapp.com/api/v1/items"
```
Comment (elijahsullivan) [inline GuildedRoseLLC/Items/RemoteItemRepository.swift:5]: I think this can be `private`. It might be worth looking through other parts of your application that are `public` when they don't need to be.

**Review by elijahsullivan — APPROVED** [review]
